### PR TITLE
first homework submission

### PR DESCRIPTION
**Pull request by diego2017**
a. 7
b.  in day 3, the bonus part of "Make it like Hangman" was a bit complex. I think we had to use many 'ifs' and loops and I wasn't able to do it.
c.  maybe just some advice on how to organise our ideas so that we know what we need to use
d. 2 - 3 hrs per day

### CODE REVIEW

**Comment by diego2017** [comment]
a. 8
b. submitting to github, but a bit.  With the for each loops I managed to used them but am not sure why it works the way it does. I guess that after doing it more times it will become easier
c. all good now
d. 3 hours

File: diego_villanueva/week01/mta_lab2/js/main.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+var createTrip = function(station1, station2, line){       //this only works within the same line
+  var position1 = findPosition(line,station1);
+  var position2 = findPosition(line,station2);
+  var direction = findDirection(position1,position2);     // this is a number
+  var trip = '';
+
+  if (direction > 0 ){       //positive, so going right
+    for (var i = position1+1; i<position2+1; i++ ){
+      window.trip.push(lines[line][i]);
+    }
```
Comment (textchimp) [inline diego_villanueva/week01/mta_lab2/js/main.js:86]: I think you'd make less work for yourself to use `.slice()` here instead of doing it the hard way with a `for` loop.

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/slice

**Comment by textchimp** [comment]
Nice work here. The bank homework was well broken down; but as we discussed, you should do some checking for overdraws in your `transfer()` function. Really nice testing/logging too! But at the same time, there's no test to check that overdraw case for transfers. Your tests should always test the fragile points in your code, places where things are most likely to go wrong: 'edge cases', bad values passed in to functions, etc...

Very decent attempt at MTA too! I wonder if it can be made any shorter?...
